### ISSUE TITLE
Edit spam

### ISSUE THREAD

**Issue by xbony2** [FTB-Gamepedia/MediaWiki-Butt-Ruby#3]
When editing, this is outputted a lot.
```
Cookie#domain returns dot-less domain name now. Use Cookie#dot_domain if you need "." at the beginning.
```

**Comment by elifoster** [comment]
Can you provide your source code? I believe that is caused by not including http:// in the initialization.

**Comment by xbony2** [comment]
```ruby
$butt = MediaWiki::Butt.new 'http://ftb.gamepedia.com'
```
From [here](https://github.com/xbony2/Experimental-Self-Aware-Electronic-Based-Space-Analyzing-Droid/blob/master/src/main/ruby/xbony2/ircbot/bonybot.rb).

**Comment by elifoster** [comment]
Weird. I remember getting that error, but it no longer happens for me.

**Comment by elifoster** [comment]
This is strange. Not much cookie stuff actually happens on our end, so it's weird that HTTPClient would be complaining about this.

See [cookie.rb](https://github.com/nahi/httpclient/blob/master/lib/httpclient/cookie.rb#L196)

**Comment by elifoster** [comment]
I'm not getting that error, even when doing -w.

Log:
```
$ ruby -w test.rb
/usr/local/lib/ruby/gems/2.2.0/gems/httpclient-2.6.0.1/lib/httpclient.rb:1083: warning: assigned but unused variable - e
/usr/local/lib/ruby/gems/2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/auth.rb:538: warning: assigned but unused variable - domain_uri
/usr/local/lib/ruby/gems/2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/webagent-cookie.rb:413: warning: private attribute?
enter password for satanicbot
```

Test script:
``` ruby
require_relative 'lib/mediawiki-butt'
require 'io/console'

$butt = MediaWiki::Butt.new("http://ftb.gamepedia.com")
$butt.logout
puts "enter password for satanicbot"
pw = STDIN.noecho(&:gets).chomp
$butt.login("SatanicBot", pw)
$butt.create_page("User:TheSatanicSanta/Sandbox/ButtTest", "stuff")
$butt.edit("User:TheSatanicSanta/Sandbox/ButtTest", "content")

$butt.logout

```

**Comment by xbony2** [comment]
TODO: [this](https://github.com/nahi/httpclient/pull/268) was merged.

**Comment by elifoster** [comment]
Should we close this due to that merged PR?

**Comment by xbony2** [comment]
Huh? Probably. Although we could probably change @@domain_warned to true to avoid seeing the warning period :P

**Comment by elifoster** [comment]
Oh, right.

**Comment by elifoster** [comment]
I guess it's not really a bug anymore...

**Comment by elifoster** [comment]
@xbony2 I haven't ever actually had a use for class variables, so how would we actually change that without inheriting HTTPClient::CookieManager? It doesn't look like it is in the client configuration stuff.

**Comment by xbony2** [comment]
Hmm? Reflection.

**Comment by elifoster** [comment]
Ah. So doing 
``` ruby
HTTPClient::WebAgent::Cookie.class_variable_set(:@@domain_warned, true)
```

just causes another warning, `warning: toplevel constant WebAgent referenced by HTTPClient::WebAgent`

**Comment by elifoster** [comment]
Fixed as of  327cfb7
